### PR TITLE
[WIP] tvm op support

### PR DESCRIPTION
**Pull request by yzhliu**

### CODE REVIEW

**Comment by TaoLv** [comment]
Nice work! BTW, can you add description? @yzhliu

**Comment by yzhliu** [comment]
@TaoLv Sure, will do.

**Comment by junrushao1994** [comment]
The build failed due to the following reason

```
Undefined symbols for architecture x86_64:
  "tvm::runtime::ExtTypeVTable::Get(int)", referenced from:
      tvm::runtime::TVMRetValue::Clear() in op_module.o
  "tvm::runtime::Registry::Get(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)", referenced from:
```

What's the cause of this? Is `TVM_DLL` defined on this function?

**Comment by marcoabreu** [comment]
Can you add cmake as well?

Also, are you sure you want to commit to the numpy branch?

**Comment by yzhliu** [comment]
@junrushao1994 weird, does this branch work? https://github.com/yzhliu/tvm-1/tree/bcast_buffer
@marcoabreu Yes cmake changes are ongoing. I'm about to open an RFC, we can definitely merge this to the master, it is not only for numpy operators.

**Comment by marcoabreu** [comment]
Great. Are you going to make the cmake changes part of this PR? I'd prefer if additions in one build system are done to the other at the same time to avoid them diverging further.

**Comment by junrushao1994** [comment]
These apis have already been declared as TVM_DLL. I have no idea why linking fails.

https://github.com/dmlc/tvm/blob/21935dcbf56ad3bd66ebff9891a6bc3865b8106d/include/tvm/runtime/packed_func.h#L434

https://github.com/dmlc/tvm/blob/21935dcbf56ad3bd66ebff9891a6bc3865b8106d/include/tvm/runtime/registry.h#L268

**Comment by yzhliu** [comment]
@marcoabreu good suggestion

**Comment by anirudhacharya** [comment]
@mxnet-label-bot add [pr-awaiting-review]

File: contrib/tvmop/opdef.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+# under the License.
+
+# coding: utf-8
+import tvm
+from itertools import product
+
+__OP_DEF__ = []
+
+class OpDef:
+    def __init__(self, func, name, target, auto_broadcast, **kwargs):
```
Comment (eric-haibin-lin) [inline contrib/tvmop/opdef.py]: The documentation should be self-contained. Could you add the doc for target, auto_broadcast, etc?

**Comment by yzhliu** [comment]
Move to https://github.com/apache/incubator-mxnet/pull/15550/files

**Pull request closed without merging**